### PR TITLE
Move `Fake` class from Mockito to the test API.

### PR DESCRIPTION
**Pull request by stereotype441**
With null safety, making a class derived from Fake will become much
more widely useful, since it will no longer be possible to for a test
method to pass `null` into a non-nullable API.

### CODE REVIEW

File: pkgs/test_api/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 0.2.20-nullsafety
```
Comment (jakemac53) [inline pkgs/test_api/CHANGELOG.md:1]: The previous version is not yet published, so you can merge this into that entry (0.2.19-nullsafety)

**Comment by jakemac53** [comment]
Did we also want to export this from a `package:test/fake.dart` library? We don't really want people to depend on test_api.

File: pkgs/test_api/CHANGELOG.md (hunk 1 above)
Comment (stereotype441) [reply pkgs/test_api/CHANGELOG.md:1]: Ok, will do.  Thanks.

**Comment by stereotype441** [comment]
> Did we also want to export this from a `package:test/fake.dart` library? We don't really want people to depend on test_api.

Sure, will do.

File: pkgs/test_api/lib/src/frontend/fake.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Extend or mixin this class to mark the implementation as a [Fake].
```
Comment (natebosch) [inline pkgs/test_api/lib/src/frontend/fake.dart:5]: Based on comments like https://github.com/dart-lang/language/issues/1068#issuecomment-656923890 I wonder if we should rephrase this to drop the encouragement of using it as a mixin. I also think it could maybe use a different phrasing in general.

```suggestion
/// A stand-in for another object which cannot be used except for specifically overridden methods.
```

Comment (stereotype441) [reply pkgs/test_api/lib/src/frontend/fake.dart]: Done

**Comment by stereotype441** [comment]
> We should also add a changelog entry into the package:test changelog (can be copy/pasted). That version is also not published so it can be added to the latest version.

Done
